colossalai/fx/tracer/tracer.py

```diff
@@ -77,18 +77,124 @@ def create_proxy(self, kind, target, args, kwargs, name=None, type_expr=None, pr
         """
         Create a proxy for different kinds of operations.
         """
-        proxy = super().create_proxy(kind, target, args, kwargs, name, type_expr, proxy_factory_fn)
 
         if self.tracer_type == TracerType.DEFAULT:
             # since meta_args is not given
             # we just fall back to the original torch.fx.Tracer
+            proxy = super().create_proxy(kind, target, args, kwargs, name, type_expr, proxy_factory_fn)
             return proxy
 
+        # if graph is traced for auto parallelism module, some extra node will be added during
+        # graph construction to deal with the compatability between bias addition and all reduce.
+
+        # if no extra manipulation is applied, we just pass the origin arguments to create_proxy function
+        # to create node on computation graph
+        origin_arguments = (kind, target, args, kwargs, name, type_expr, proxy_factory_fn)
+        # dispatch the arguments generator depending on the kind and target in origin arguments.
+        args_metas, _ = extract_meta(*args, **kwargs)
+        if kind == "call_function":
+            if bias_addition_function.has(target):
+                return bias_addition_function.get(target)
+            elif bias_addition_function.has(target.__name__):
+                # use name for some builtin op like @ (matmul)
+                return bias_addition_function.get(target.__name__)
+
+        elif kind == "call_method":
+            method = getattr(args_metas[0].__class__, target)
+            if bias_addition_function.has(method):
+                proxy_arguments_dict = bias_addition_function.get(method)
```

Comment:
don't quite get it why it directly returns when calling `call_function` but does not return for `call_method`.

Reply:
Sorry, I will fix it, I just focus on the call_module cases last night